data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/OperatorFactory.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.expression;
+
+import org.opensearch.dataprepper.expression.antlr.DataPrepperExpressionParser;
+import org.springframework.context.annotation.Bean;
+
+import javax.inject.Named;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+
+@Named
+public class OperatorFactory {
+    public final BiPredicate<Object, Object> regexEquals = (x, y) -> ((String) x).matches((String) y);
+    public final BiPredicate<Object, Object> equals = Objects::equals;
+    public final BiPredicate<Object, Object> inSet = (x, y) -> ((Set<?>) y).contains(x);
+
+    @Bean
+    public NumericCompareOperator greaterThanOperator() {
+        final Map<Class<? extends Number>, Map<Class<? extends Number>, BiFunction<Object, Object, Boolean>>>
```

Comment:
I'd recommend having made a `GreaterThanOperator` which inherits from `NumericCompareOperator`. It would provide these functions in the constructor. That would keep this class straightforward and simple to test.

```
class GreaterThanOperator {
  private GreaterThanOperator(Map<Class<? extends Number>, Map<Class<? extends Number>, BiFunction<Object, Object, Boolean>>> operandsToOperationMap) {
   super(DataPrepperExpressionParser.GT, operandsToOperationMap);
  }

  public static GreaterThanOperator create {
    /* All the code currently in greaterThanOperator() */
    return new GreaterThanOperator(operandsToOperationMap);
  }
}
```

This is all package protected code, so I'm fine with what is here though.

Reply:
I think the pattern you suggest is closer to the following:
```
@Named
class GreaterThanOperator extends NumericCompareOperator{
  
   @Inject // @Inject probably not necessary
   public GreaterThanOperator() {
      Map<Class<? extends Number>, Map<Class<? extends Number>, BiFunction<Object, Object, Boolean>>> operandsToOperationMap = ...
       super(DataPrepperExpressionParser.GT, operandsToOperationMap);
   }
}
```
TBH, I do not know from DI point view which pattern is favored, and there maybe other reasonings. We kind of already swang back and forth here. Since it is currently not blocking, I will go ahead and merge this PR.

Reply:
You actually cannot do what you have above because `super` must be the first line in the constructor. You could add a static method which creates the `Map` and have `super(DataPrepperExpressionParser.GT, createOperationsMap());`. Or take the approach I had.